core/mavlink_system.cpp

```diff
@@ -282,16 +295,82 @@ void Device::device_thread(Device *self)
     }
 }
 
-void Device::send_heartbeat(Device &self)
+std::string MAVLinkSystem::component_name(uint8_t component_id)
+{
+    switch (component_id) {
+        case MAV_COMP_ID_AUTOPILOT1:
+            return "Autopilot";
+        case MAV_COMP_ID_CAMERA:
+            return "Camera 1";
+        case MAV_COMP_ID_CAMERA2:
+            return "Camera 2";
+        case MAV_COMP_ID_CAMERA3:
+            return "Camera 3";
+        case MAV_COMP_ID_CAMERA4:
+            return "Camera 4";
+        case MAV_COMP_ID_CAMERA5:
+            return "Camera 5";
+        case MAV_COMP_ID_CAMERA6:
+            return "Camera 6";
+        case MAV_COMP_ID_GIMBAL:
+            return "Gimbal";
+        default:
+            return "Unsupported component";
+    }
+}
+
+void MAVLinkSystem::add_new_component(uint8_t component_id)
+{
+    auto res_pair = _components.insert(component_id);
+    if (res_pair.second) {
+        LogDebug() << "Component " << component_name(component_id) << " added.";
+    }
+}
+
+size_t MAVLinkSystem::total_components() const
+{
+    return _components.size();
+}
+
+bool MAVLinkSystem::is_standalone() const
+{
+    return !is_autopilot();
+}
+
+bool MAVLinkSystem::is_autopilot() const
```

Comment:
I would change this to `has_autopilot()` to make it consistent and because it's not clear if something **is** an autopilot if it also includes a camera.

Reply:
_(FYI: I am unable to use Slack temporarily, but I see your email notifications.)_

OK. I added `is_autopilot` thinking that we could say,
```cpp
...
if (system.is_autopilot() && system.has_camera())
    print "We've a drone that has a camera"
else if (system.is_autopilot())
    print "We've a drone"
...
```
I understand your point as well, so I am ok to change as you suggested.
I am also thinking to add `is_camera()` for standalone cameras. 
If something **is** a camera, it can't be anything else; because, it is drones that contain camera and other components. What do you think ?

Reply:
I don't think we need `is_camera()` because it will be confusing to have both. And by just having `has_...()` it is clear that a system can always include various components.